source/docs/software/telemetry/writing-sendable-classes.rst

```diff
@@ -0,0 +1,71 @@
+Writing Your Own Sendable Classes
+=================================
+
+Since the ``Sendable`` interface only has one method, writing your own classes that implement ``Sendable`` (and thus automatically log values to and/or consume values from the dashboard) is extremely easy: just provide an implementation for the overrideable ``initSendable`` method, in which setters and getters for your class's fields are declaratively bound to key values (their display names on the dashboard).
+
+For example, here is the implementation of ``initSendable`` from WPILib's ``BangBangController``:
+
+.. tabs::
+
+    .. group-tab:: Java
+
+        .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/main/wpimath/src/main/java/edu/wpi/first/math/controller/BangBangController.java
+           :language: java
+           :lines: 150-158
+           :linenos:
+           :lineno-start: 150
+
+    .. group-tab:: C++
+
+        .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/main/wpimath/src/main/native/cpp/controller/BangBangController.cpp
+           :language: cpp
+           :lines: 58-72
+           :linenos:
+           :lineno-start: 58
+
+In order to enable the automatic updating of values by WPILib "in the background", ``Sendable`` data names are bound to getter and setter functions rather than specific data values.  If a field that you wish to log has no defined setters and getters, they can be defined inline with a lambda expression.
```

Comment:
```suggestion
To enable the automatic updating of values by WPILib "in the background", ``Sendable`` data names are bound to getter and setter functions rather than specific data values.  If a field that you wish to log has no defined setters and getters, they can be defined inline with a lambda expression.
```